### PR TITLE
chore(ios): removed custom asset image hashing

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-25032

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -5768,14 +5768,29 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
 						imageNameRegExp = /^(.*?)(-dark)?(@[23]x)?(~iphone|~ipad)?\.(png|jpg)$/;
 
 					Object.keys(imageAssets).forEach(function (file) {
+						const directories = file.split('/');
+						let newPath = '';
+						for (let i = 0; i < directories.length - 1; i++) {
+							newPath = newPath + '/' + directories[i];
+
+							writeAssetContentsFile.call(this, path.join(assetCatalog, newPath, 'Contents.json'), {
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Are you sure you mean to create a separate `Contents.json` file in each directory?

Also, this code will recreate the file for higher up directories. In other words `foo/bar/baz.png` and `foo/wiz/pow.png` will create `foo/Contents.json` twice. Seems unnecessary.

Comment (vijaysingh-axway) [reply iphone/cli/commands/_build.js]: 1. Yes, Content.json is required in each directory.
2. I believe condition "if (!fs.existsSync(dest) || contents !== fs.readFileSync(dest).toString())" in function "writeAssetContentsFile(dest, json)" will ensure not to write Content.js json twice in same directory.

Comment (cb1kenobi) [reply iphone/cli/commands/_build.js]: Ah, I see now. Make sense. Thanks!

**Review by cb1kenobi — APPROVED** [review]
CR'd the build changes and they look good to me. I have not performed a FR.

**Review by janvennemann — APPROVED** [review]
Native iOS changes are looking good as well!

**Comment by ssaddique** [comment]
FR Passed.

Test environment:
OS Ver:         10.14.6
Xcode Ver:      Xcode 11.3.1
Appc NPM:       4.2.15
Appc CLI:       7.1.2
Ti CLI Ver:     5.2.2
Alloy Ver:      1.14.4
Node Ver:       10.17.0
NPM Ver:        6.13.6
Emulators:      iPhone 7 (iOS 10), iPhone X (iOS 11), iPhone XS (iOS 12), iPhone 11 Pro (iOS 13)

**Comment by hansemannn** [comment]
This was merged way too early, sorry guys. New regressions:
- Sub folders are not properly handled anymore
- Unrelated folders are included in the asset catalog although not inside `Resources/iphone` (e.g. `Resources/test`)

**Comment by hansemannn** [comment]
This was merged way too early, sorry guys. New regressions:
- Sub folders are not properly handled anymore
- Unrelated folders are included in the asset catalog although not inside `Resources/iphone` (e.g. `Resources/test`)

**Comment by vijaysingh-axway** [comment]
> This was merged way too early, sorry guys. New regressions:
> 
> * Sub folders are not properly handled anymore
> * Unrelated folders are included in the asset catalog although not inside `Resources/iphone` (e.g. `Resources/test`)

@hansemannn  Can you provide test case please?
@lokeshchdhry Can you verify with more test cases apart from mentioned in ticket.

**Comment by hansemannn** [comment]
@vijaysingh-axway Interesting edge case actually. What if a user wants to use a raw image that shouldn't go to the asset catalog? It's definitely no show stopper, but may get back to the universal-asset-catalog-masterpiece from @cb1kenobi again 😛(happy birthday btw, man!)
